app/server/src/main/scala/org/bitcoins/server/Main.scala

```diff
@@ -28,21 +29,41 @@ import scala.concurrent.{ExecutionContext, Future, Promise}
 object Main extends App with BitcoinSLogger {
 
   private def runMain(): Unit = {
-    implicit val system: ActorSystem = ActorSystem("bitcoin-s")
-    implicit val ec: ExecutionContext = system.dispatcher
     val argsWithIndex = args.zipWithIndex
 
-    implicit val conf: BitcoinSAppConfig = {
+    val dataDirIndexOpt = {
+      argsWithIndex.find(_._1.toLowerCase == "--datadir")
+    }
+    val datadirPath = dataDirIndexOpt match {
+      case None => AppConfig.DEFAULT_BITCOIN_S_DATADIR
+      case Some((_, dataDirIndex)) =>
+        val str = args(dataDirIndex + 1)
+        Paths.get(str)
+    }
 
-      val dataDirIndexOpt = {
-        argsWithIndex.find(_._1.toLowerCase == "--datadir")
-      }
-      val datadirPath = dataDirIndexOpt match {
-        case None => AppConfig.DEFAULT_BITCOIN_S_DATADIR
-        case Some((_, dataDirIndex)) =>
-          val str = args(dataDirIndex + 1)
-          Paths.get(str)
+    val baseConfig = AppConfig.getBaseConfig(datadirPath)
+
+    val networkStr = baseConfig.getString("bitcoin-s.network")
+    val network = BitcoinNetworks.fromString(networkStr)
+
+    val datadir: Path = {
+      val lastDirname = network match {
+        case MainNet  => "mainnet"
+        case TestNet3 => "testnet3"
+        case RegTest  => "regtest"
       }
+      datadirPath.resolve(lastDirname)
+    }
+
+    System.setProperty("bitcoins.log.location",
```

Comment:
So i'm guessing this isn't going to work if we use the `node` project as a library?

Reply:
The bug was independent of the `node` project since it takes in its own `ActorSystem`. If someone else if using the `node` module in their project they'll have their own `logback.xml`  and `ActorSystem` to configure for however they want to do logging.